Source/Scene/Points3DTileContent.js

```diff
@@ -164,159 +196,366 @@ define([
+        }
+        //>>includeEnd('debug');
         byteOffset += sizeOfUint32;
 
-        var batchTableJSONByteLength = view.getUint32(byteOffset, true);
+        var featureTableBinaryByteLength = view.getUint32(byteOffset, true);
         byteOffset += sizeOfUint32;
 
-        var batchTableBinaryByteLength = view.getUint32(byteOffset, true);
-        byteOffset += sizeOfUint32;
+        var featureTableString = getStringFromTypedArray(uint8Array, byteOffset, featureTableJSONByteLength);
+        var featureTableJSON = JSON.parse(featureTableString);
+        byteOffset += featureTableJSONByteLength;
 
-        var positions = new Float32Array(arrayBuffer, byteOffset, pointsLength * 3);
-        byteOffset += pointsLength * 3 * sizeOfFloat32;
+        var featureTableBinary = new Uint8Array(arrayBuffer, byteOffset, featureTableBinaryByteLength);
+        byteOffset += featureTableBinaryByteLength;
 
-        var colors;
-        var translucent = false;
-        var hasConstantColor = false;
-
-        if (batchTableJSONByteLength > 0) {
-            // Get the batch table JSON
-            var batchTableString = getStringFromTypedArray(uint8Array, byteOffset, batchTableJSONByteLength);
-            var batchTableJSON = JSON.parse(batchTableString);
-            byteOffset += batchTableJSONByteLength;
-
-            // Get the batch table binary
-            var batchTableBinary;
-            if (batchTableBinaryByteLength > 0) {
-                batchTableBinary = new Uint8Array(arrayBuffer, byteOffset, batchTableBinaryByteLength);
-            }
-            byteOffset += batchTableBinaryByteLength;
-
-            // Get the point colors
-            var tiles3DRGB = batchTableJSON.TILES3D_RGB;
-            var tiles3DRGBA = batchTableJSON.TILES3D_RGBA;
-            var tiles3DColor = batchTableJSON.TILES3D_COLOR;
-
-            if (defined(tiles3DRGBA)) {
-                colors = new Uint8Array(batchTableBinary, tiles3DRGBA.byteOffset, pointsLength * 4);
-                translucent = true;
-            } else if (defined(tiles3DRGB)) {
-                colors = new Uint8Array(batchTableBinary, tiles3DRGB.byteOffset, pointsLength * 3);
-            } else if (defined(tiles3DColor)) {
-                this._constantColor = Color.fromBytes(tiles3DColor[0], tiles3DColor[1], tiles3DColor[2], tiles3DColor[3], this._constantColor);
-                hasConstantColor = true;
+        this._featureTableResources = new Cesium3DTileFeatureTableResources(featureTableJSON, featureTableBinary);
+
+        this.state = Cesium3DTileContentState.PROCESSING;
+        this.contentReadyToProcessPromise.resolve(this);
+    };
+
+    function createResources(content, frameState) {
+        var context = frameState.context;
+        var featureTableResources = content._featureTableResources;
+        var featureTableJSON = featureTableResources.json;
+
+        var pointsLength = featureTableResources.getGlobalProperty('POINTS_LENGTH');
```

Comment:
Instead of using `_featureTableResources` as an intermediate object between parsing and resource creation, what do you think about introduce a new, local class so that you could parse and validate right away in `initialize` and then create resources here?  It may be a bit more code, but it is a more cohesive design (i.e., separation of parsing and resource creation, and throws `DeveloperError`s consistently from the same place).